File: dotnet/Applications/GameX.App.Explorer/Controls/ViewTestTri.cs
using OpenStack.Gfx;
using OpenStack.Gfx.Gl;

namespace GameX.App.Explorer.Controls
{
    public class ViewTestTri : ViewBase<object>
    {
        //protected override void SetViewportSize(int x, int y, int width, int height)
        //{
        //    if (Obj == null) return;
        //    if (Obj.Width > 1024 || Obj.Height > 1024 || false) base.SetViewportSize(x, y, width, height);
        //    else base.SetViewportSize(x, y, Obj.Width << FACTOR, Obj.Height << FACTOR);
        //}

        protected override (object, IList<IRenderer>) GetObj(object source)
        {
            var obj = source;
            return (obj, [new TestTriRenderer(GL)]);
        }
    }
}

File: python/app/widgets/ViewTestTri.py
import sys, os
import numpy as np
from OpenGL.GL import *
from openstk.gfx.gfx_render import IRenderer
from openstk.gfx.gl_renderer import TestTriRenderer
from .ViewBase import ViewBase

# ViewTestTri
class ViewTestTri(ViewBase):
    def __init__(self, parent, tab):
        super().__init__(parent, tab)

    def getObj(self, source: object) -> (object, list[IRenderer]):
        obj: object = self.obj
        return (obj, [TestTriRenderer(self.gl)])
